apps/examples/ble_perfs/ble_perfs_main.c

```diff
@@ -0,0 +1,452 @@
+};
+
+
+static char * attr_cb_type_str[] = {
+	"BLE_SERVER_ATTR_CB_WRITING",
+	"BLE_SERVER_ATTR_CB_READING",
+	"BLE_SERVER_ATTR_CB_WRITING_NO_RSP",
+};
+
+static char * connection_type_str[] = {
+	"BLE_SERVER_LL_CONNECTED",
+	"BLE_SERVER_SM_CONNECTED",
+	"BLE_SERVER_DISCONNECTED",
+};
+
+static char * get_attr_cb_type_str(ble_server_attr_cb_type_e cb_type)
+{
+	return  attr_cb_type_str[cb_type];
+};
+
+
+static char * get_conn_type_str(ble_server_connection_type_e conn_type)
+{
+	return connection_type_str[conn_type];
+}
+
+static void addr_string_to_type(uint8_t *addr, const char *address)
+{
+	int i;
+	char *ptr = NULL;
+
+	for (i = 0; i < BLE_BD_ADDR_MAX_LEN; i++) {
+		addr[i] = strtol(address, &ptr, 16);
+		if (ptr != NULL) {
+			if (ptr[0] != ':')
+				return;
+			address = ptr + 1;
+		}
+	}
+}
+
+static void addr_type_to_string(char *address, const uint8_t *addr)
+{
+	if (addr == NULL) {
+		strncpy(address, "NULL", BLE_BD_ADDR_STR_LEN);
+		return;
+	}
+
+	snprintf(address, BLE_BD_ADDR_STR_LEN + 1, "%02x:%02x:%02x:%02x:%02x:%02x",
+			addr[0], addr[1], addr[2], addr[3], addr[4], addr[5]);
+
+}
+
+
+static void utc_cb_charact_a_1(ble_server_attr_cb_type_e type, ble_conn_handle conn_handle, ble_attr_handle attr_handle, void *arg)
+{
+	char *arg_str = "None";
+
+	if (arg != NULL) {
+		arg_str = (char *)arg;
+	}
+
+	BLE_LOGD("[CHAR_A_1][%s] type : %d / handle : %d / attr : %02x \n", arg_str, type, conn_handle, attr_handle);
+}
+
+static void utc_cb_desc_b_1(ble_server_attr_cb_type_e type, ble_conn_handle conn_handle, ble_attr_handle attr_handle, void *arg)
+{
+	char *arg_str = "None";
+	if (arg != NULL) {
+		arg_str = (char *)arg;
+	}
+	BLE_LOGD( "[DESC_A_1][%s] type : %d / handle : %d / attr : %02x \n", arg_str, type, conn_handle, attr_handle);
+}
+
+
+static void perfs_ble_reset_data(void)
+{
+	g_perf_data.packet_size = 0;
+	g_perf_data.packet_count = 0;
+
+	g_perf_data.start_time.tv_sec = 0;
+	g_perf_data.start_time.tv_usec = 0;
+	g_perf_data.end_time.tv_sec = 0;
+	g_perf_data.end_time.tv_usec = 0;
+}
+
+static int perfs_ble_get_result(void)
+{
+	struct timeval diff_time = {0, 0};
+	double total_time = 0;
+	double speed = 0;
+
+	BLE_LOGD( "======== Bandwidth Analysis =================\n");
+        BLE_LOGD( "start time: %ld sec %ld usec \n", g_perf_data.start_time.tv_sec, g_perf_data.start_time.tv_usec);
+        BLE_LOGD( "end time: %ld sec %ld usec \n", g_perf_data.end_time.tv_sec, g_perf_data.end_time.tv_usec);
+
+	diff_time.tv_sec = g_perf_data.end_time.tv_sec - g_perf_data.start_time.tv_sec;
+	diff_time.tv_usec = g_perf_data.end_time.tv_usec - g_perf_data.start_time.tv_usec;
+	if(diff_time.tv_usec < 0) {
+		diff_time.tv_sec = diff_time.tv_sec - 1;
+		diff_time.tv_usec = diff_time.tv_usec + 1000000;
+	}
+
+        BLE_LOGD( "diff time: %ld sec %ld usec \n", diff_time.tv_sec, diff_time.tv_usec);
+	
+	total_time = diff_time.tv_sec + (double) diff_time.tv_usec / 1000000;
```

Comment:
Please remove the empty space after type casting.

Reply:
done